Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -406,50 +406,46 @@ public BindingPair(BindingExpressionPart part, object source, bool isLast)
 			public object Source { get; private set; }
 		}
 
-		class WeakPropertyChangedProxy
+		internal class WeakPropertyChangedProxy
 		{
-			WeakReference _source, _listener;
-			internal WeakReference Source => _source;
+			readonly WeakReference<INotifyPropertyChanged> _source = new WeakReference<INotifyPropertyChanged>(null);
+			readonly WeakReference<PropertyChangedEventHandler> _listener = new WeakReference<PropertyChangedEventHandler>(null);
+			readonly PropertyChangedEventHandler _handler;
+			internal WeakReference<INotifyPropertyChanged> Source => _source;
 
-			public WeakPropertyChangedProxy(INotifyPropertyChanged source, PropertyChangedEventHandler listener)
+			public WeakPropertyChangedProxy()
```

Comment:
the purpose of these changes is to be able to reuse the WPCP and the underlying WeakReferences, thus avoiding allocations